stream_consumer.go

```diff
@@ -347,6 +351,38 @@ func (se *streamEndpoint) reconnectWhileNotStopped(c *consumer, streamName strin
 	}
 }
 
+type connectionStatus int
+
+const (
+	connected connectionStatus = iota
+	notConnected
+	closed
+)
+
+func (se *streamEndpoint) waitForHelloMessage(c *consumer, streamName string, st stream.Stream_StreamClient) connectionStatus {
+	Log.Debug("Waiting for Hello message", zap.String("stream", streamName), zap.String("target", se.target))
```

Comment:
does it make sense to wait a max amount of time?

Reply:
it would be nice but the call to receive does not take a context, so this would complicate the code